src/hotspot/share/utilities/vmError.cpp

```diff
@@ -1678,7 +1710,17 @@ void VMError::report_and_die(int id, const char* message, const char* detail_fmt
     OnError = nullptr;
   }
 
-  if (WINDOWS_ONLY(!UseOSErrorReporting) NOT_WINDOWS(true)) {
+  if (after_report_action == AfterReportAction::Return) {
+    // Caller requested reporting without termination.
+    return;
+  } else if (after_report_action == AfterReportAction::Die) {
+#ifdef _WINDOWS
+    if (UseOSErrorReporting) {
```

Comment:
Since you already have to use UseOSErrorReporting in this code, I don't see the purpose of the parameter and ARA code.  If you add a 'return' here and have the other code snippet do a return, this should be all you need.
I'm not convinced that the DebugBreak here for windows is useful, since if you want to attach to a debugger, you'd use ShowMessageBoxOnError (based on my sample size of  1).  But moving it here away from the vmassert macro seems fine.  I tested that this works.
When implementing WER, they'll want a non-empty mdmp file but that can probably be done with call to the minidump facility which can be factored out of the os::abort() call.
My original interest is to get meaningful mdmp files showing the correct stack trace of the crashing thread.  This doesn't break or fix that.